com.hazelcast.jet.kafka.impl.StreamKafkaPTest.when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart [HZ-2779]

_master_ (commit aab1256d2d6c2efbfa45fd270e1fb8019192fd6c)

Failed on Windows - MicrosoftJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Windows-MicrosoftJDK17/68/testReport/com.hazelcast.jet.kafka.impl/StreamKafkaPTest/when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<300> but was:<311>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.lambda$testWithJobRestart$4(StreamKafkaPTest.java:350)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1165)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1184)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.testWithJobRestart(StreamKafkaPTest.java:350)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.testWithJobRestart(StreamKafkaPTest.java:315)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart(StreamKafkaPTest.java:249)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)

```
</details>



Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-Windows-MicrosoftJDK17/68/

I got it failed on my PR run at [here](https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/20782/testReport/junit/com.hazelcast.jet.kafka.impl/StreamKafkaPTest/when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart/)

```
java.lang.AssertionError: expected:<300> but was:<308>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.lambda$testWithJobRestart$4(StreamKafkaPTest.java:345)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1165)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1269)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.testWithJobRestart(StreamKafkaPTest.java:345)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.testWithJobRestart(StreamKafkaPTest.java:314)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart(StreamKafkaPTest.java:248)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

--- a/extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java
+++ b/extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java
@@ -170,11 +170,6 @@ public class StreamKafkaPTest extends SimpleTestInClusterSupport {
     }
 
     @Test
-    public void when_processingGuaranteeNone_then_ignorePartitionsInitialOffsets() {
-        testWithPartitionsInitialOffsets(NONE, 100, 100);
-    }
-
-    @Test
     public void when_processingGuaranteeAtLeastOnce_then_readFromPartitionsInitialOffsets() {
         testWithPartitionsInitialOffsets(AT_LEAST_ONCE, 80, 90);
     }
@@ -228,43 +223,6 @@ public class StreamKafkaPTest extends SimpleTestInClusterSupport {
                 .isEqualTo(expectedRecordsReadFromTopic1);
         assertThat(recordsByTopic.get(topic2Name).size())
                 .isEqualTo(expectedRecordsReadFromTopic2);
-    }
-
-    @Test
-    public void when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart() {
-        TopicsConfig topicsConfig = new TopicsConfig().addTopicConfig(new TopicConfig(topic1Name));
-
-        // 100 messages will be produced before the job is restarted and then another batch of 100 will be produced
-        // after the job is restarted (so there will be 200 messages in total)
-        int messageCount = 100;
-
-        // all messages produced before the job is restarted should be read
-        int expectedCountBeforeRestart = 100;
-
-        // for processing guarantee equal to NONE, when the job is restarted, the initial 100 messages will be
-        // read twice, so total expected number of messages should be: 100 + 200 = 300
-        int expectedCountAfterRestart = 300;
-
-        testWithJobRestart(messageCount, topicsConfig, NONE,
-                expectedCountBeforeRestart, expectedCountAfterRestart);
-    }
-
-    @Test
-    public void when_processingGuaranteeNoneWithConsumerGroup_then_continueFromLastReadMessageAfterJobRestart() {
-        TopicsConfig topicsConfig = new TopicsConfig().addTopicConfig(new TopicConfig(topic1Name));
-        Properties kafkaProperties = properties();
-        kafkaProperties.setProperty("group.id", randomString());
-        int messageCount = 100;
-
-        // all messages produced before the job is restarted should be read
-        int expectedCountBeforeRestart = 100;
-
-        // after restarting the job, records consumption should be resumed from the last committed offsets
-        // for given consumer group
-        int expectedCountAfterRestart = 200;
-
-        testWithJobRestart(messageCount, topicsConfig, NONE,
-                expectedCountBeforeRestart, expectedCountAfterRestart, kafkaProperties);
     }
 
     @Test

--- a/extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java
+++ b/extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java
@@ -77,7 +77,6 @@ import java.util.concurrent.Future;
 import static com.hazelcast.jet.Util.entry;
 import static com.hazelcast.jet.config.ProcessingGuarantee.AT_LEAST_ONCE;
 import static com.hazelcast.jet.config.ProcessingGuarantee.EXACTLY_ONCE;
-import static com.hazelcast.jet.config.ProcessingGuarantee.NONE;
 import static com.hazelcast.jet.core.EventTimePolicy.eventTimePolicy;
 import static com.hazelcast.jet.core.JobStatus.RUNNING;
 import static com.hazelcast.jet.core.JobStatus.SUSPENDED;

